prov/rxm/src/rxm_init.c

```diff
@@ -412,6 +429,11 @@ RXM_INI
 			"decrease noise during cq polling, but may result in "
 			"longer connection establishment times. (default: 10000).");
 
+	fi_param_define(&rxm_prov, "data_progress_auto", FI_PARAM_BOOL,
+			"Set default data progress mode to FI_PROGRESS_AUTO "
+			"(default: 0). Application hints takes precedence "
+			"over this.\n");
```

Comment:
Should the application hints take precedence, or do we want to force auto progress under the covers?

Reply:
I think this env variable should change only the default progress mode (when app doesn't care). If an app explicitly requires a progress mode, we can't return a different mode. Middleware like MPI just pick whatever progress mode the provider reports. So this env variable can help test different progress modes.

Although in mercury RPC's case, this variable would not be useful as their app doesn't set the progress to FI_PROGRESS_UNSPEC if I remember correctly.

Reply:
I don't know that changing the default return value is that useful.  We can test different modes by controlling the value in the test app.  For the Mercury problem, I think we need to enable auto progress, separate from what we actually report. That's what I thought your intent was.

Reply:
Does the API allow a provider to return auto progress when hints specify manual progress?

Changing the test app is not easy unless the app provides an env variable / command line option to do that.

I was hoping this might help mercury but on double-checking mercury actually specifies a progress mode explicitly in hints. So mercury app developers may need to provide a way to change it.